nbconvert/exporters/export.py

```diff
@@ -165,30 +183,30 @@ def export_by_name(format_name, nb, **kw):
         Name of the template style to export to.
     """
     
-    exporter = get_exporter(name)
-    function_name = "export_" + format_name.lower()
-    
-    if function_name in globals():
-        return globals()[function_name](nb, **kw)
-    else:
-        exporter = get_exporter(name)
-        
-        raise ExporterNameError("Exporter for `%s` not found" % function_name)
+    warnings.warn("export_by_name is deprecated since nbconvert 5.0", DeprecationWarning, stacklevel=3)
```

Comment:
Won't this warn twice, because you're also warning in the decorator? Also, I think `stacklevel` should be 2, as it often is.